cobc/codegen.c

```diff
@@ -7099,29 +7099,29 @@ output_field_constant (cb_tree x, int n, const char *flagname)
 	output_newline ();
 }
 
-static void output_java_call(const char *p)
+static void
+output_java_call (struct cb_call *p)
 {
+	char* class_and_method_name = p->name + 5; /* Assume java.prefix (enforced in `parser.y`, rule `call_body`)*/
 	char *last_dot;
 	char *method_name;
 	const char *class_name;
 	char* duplicate;
-	char transformed_p[strlen(p) + 1];
-
-	old_p = strdup(p);
+	char transformed_p[strlen(class_and_method_name) + 1];
 
-	for (size_t i = 0; i < strlen(p) + 1; i++) {
-		transformed_p[i] = (p[i] == '.') ? '_' : p[i];
+	for (size_t i = 0; i < strlen(class_and_method_name) + 1; i++) {
+		transformed_p[i] = (class_and_method_name[i] == '.') ? '_' : class_and_method_name[i];
 	}
 
-	last_dot = strrchr(p, '.');
+	last_dot = strrchr(class_and_method_name, '.');
 	if (last_dot == NULL) {
-		cobc_err_msg (_("malformed call '%s' to a Java method"), p);
+		cobc_err_msg (_("malformed call '%s' to a Java method"), class_and_method_name);
```

Comment:
For later: I think this check should actually end up (also) earlier during parsing.